test/integration/Session.test.js

```diff
@@ -0,0 +1,89 @@
+import assert from 'assert'
+
+import StreamrClient from '../../src'
+import config from './config'
+
+describe('Session', () => {
+    let clientApiKey
+    let clientPrivateKey
+    let clientUsernamePassword
+    let clientNone
+
+    const createClient = (opts = {}) => new StreamrClient({
+        url: config.websocketUrl,
+        restUrl: config.restUrl,
+        autoConnect: false,
+        autoDisconnect: false,
+        ...opts,
+    })
+
+    beforeAll(() => {
+        clientApiKey = createClient({
+            auth: {
+                apiKey: 'tester1-api-key',
+            },
+        })
+        clientPrivateKey = createClient({
+            auth: {
+                privateKey: '348ce564d427a3311b6536bbcff9390d69395b06ed6c486954e971d960fe8709',
+            },
+        })
+        clientUsernamePassword = createClient({
+            auth: {
+                username: 'tester2@streamr.com',
+                password: 'tester2',
+            },
+        })
+        clientNone = createClient({
+            auth: {},
+        })
+    })
+
+    describe('Token retrievals succeed', () => {
+        it('should get token from API key', () => clientApiKey.session.getSessionToken()
+            .then((sessionToken) => {
+                assert(sessionToken)
+            }))
+        it('should get token from private key', () => clientPrivateKey.session.getSessionToken()
+            .then((sessionToken) => {
+                assert(sessionToken)
+            }))
+        it('should get token from username/password', () => clientUsernamePassword.session.getSessionToken()
+            .then((sessionToken) => {
+                assert(sessionToken)
+            }))
+        it('should fail to get token with no authentication', (done) => clientNone.session.getSessionToken()
+            .catch((err) => {
+                assert.equal(err.toString(), 'Error: Need either "privateKey", "apiKey" or "username"+"password" to login.')
+                done()
+            }))
+    })
+
+    describe('Internal state', () => {
```

Comment:
This stuff would be better suited for a unit test (see comment in `Session.js`, and earlier comment about testing the `loginFunction` calls). Then you can mock the collaborator calls (under `_client`) with `sinon.mock()`s, making it very easy to test that `Session` reacts appropriately.